bugsnag-plugin-android-ndk/src/main/jni/deps/parson/parson.c

```diff
@@ -998,6 +1000,20 @@ static int json_serialize_to_buffer_r(const JSON_Value *value, char *buf, int le
             }
             written_total += written;
             return written_total;
+        case JSONLong:
+            l_value = json_value_get_long(value);
+            if (buf != NULL) {
+                num_buf = buf;
+            }
+            written = sprintf(num_buf, "\"%lld\"", (long long) l_value);
```

Comment:
Provided you import [`<inttypes.h>`](https://www.cplusplus.com/reference/cinttypes/), you can do:

```suggestion
            written = sprintf(num_buf, "\"%" PRId64 "\"", l_value);
```

and you'll get the right number type on each arch.

Regarding stringification, [RFC 7159](https://www.rfc-editor.org/rfc/rfc7159#section-6) states that JSON number values should be recognized correctly within the range +/- 2^53 (exclusive):

> Note that when such software is used, numbers that are integers and
   are in the range [-(2\**53)+1, (2\**53)-1] are interoperable in the
   sense that implementations will agree exactly on their numeric
   values.

So if you want a smaller change footprint (e.g. `versionCode` will still be a number for people until they get to 2 ^ 1023), you can check if a value is inside of the range and only stringify if not:

```c
if (l_value =< MAX_NUM || l_value >= MIN_NUM) { // not the best constant names, but you know
  written = sprintf(num_buf, "%" PRId64, l_value);
} else {
  written = sprintf(num_buf, "\"%" PRId64 "\"", l_value);
}
```